ipv-core/src/main/java/com/ibm/whc/deid/util/RandomGenerators.java

```diff
@@ -260,23 +262,40 @@ public static String generateRandomURL() {
   }
 
   /**
-   * Random date milliseconds long.
-   *
-   * @return the long
+   * @return a random date before the current year
    */
-  public static long randomDateMilliseconds() {
-    long currentMillis = System.currentTimeMillis();
-    return currentMillis - (long) random.nextInt(100) * 365 * 24 * 60 * 60 * 1000;
-  }
-
-  /**
-   * Generate random date string.
-   *
-   * @return the string
-   */
-  public static String generateRandomDate() {
-    Date date = new Date(randomDateMilliseconds());
-    return date.toString();
+  public static OffsetDateTime generateRandomDate() {
+    LocalDateTime now = LocalDateTime.now();
+    int yearRange = now.get(ChronoField.YEAR) - 1900;
+    int month = random.nextInt(12) + 1;
+    int dayRange;
+    switch (month) {
+      case 1:
+      case 3:
+      case 5:
+      case 7:
+      case 8:
+      case 10:
+      case 12:
+        dayRange = 31;
+        break;
+      case 4:
+      case 6:
+      case 9:
+      case 11:
+        dayRange = 30;
+        break;
+      case 2:
+        dayRange = 28;

```

Comment:
Is there a need to be able to generate leap year (Feb 29)?  I think the old code can potentially generate Feb 29.  Not sure if we need to be fancy and have 1/4 chance of setting dayRange to 29 for Feb.

Reply:
May not hurt to have a small chance of having it. We found bugs that way already.